contrib/native/client/CMakeLists.txt

```diff
@@ -88,19 +88,28 @@ if(MSVC)
     set(Boost_USE_STATIC_LIBS ON)
     set(Boost_USE_MULTITHREADED ON)
     set(Boost_USE_STATIC_RUNTIME OFF)
+elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    set(Boost_USE_STATIC_LIBS ON)
+    set(Boost_USE_MULTITHREADED ON)
+    #    To build a production version, the macos build must use a shaded version
+    #    of boost. Arbirtarily, we choose the new namspace to be drill_boost.
+    #    See the instructions in the readme for the mac and rebuild boost. Then
+    #    uncomment the line below to build
+    #    set(Boost_NAMESPACE drill_boost)
```

Comment:
if the boost namespace is changed, I guess the code should be changed too? But it means that all boost symbols are changed too (included boost::shared_ptr<> which are part of the public API), correct?

In any case, can you also provide the changes to the source files?